independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -110,7 +110,7 @@ static BeanInfo createClassBean(ClassInfo beanClass, BeanDeployment beanDeployme
 
         BeanInfo bean = new BeanInfo(beanClass, beanDeployment, scope, types, qualifiers,
                 Injection.forBean(beanClass, null, beanDeployment, transformer), null, null,
-                isAlternative ? alternativePriority : null, stereotypes, name);
+                isAlternative ? alternativePriority : null, stereotypes, name, false);
```

Comment:
I believe this should work for class beans as well...

Reply:
I think it wouldn't hurt, but I am having a hard time envisioning a scenario where someone would create a producer method and also annotate the class itself as well. What do you think?

Reply:
I have left it as is for now since I think that it's the most common case. We can always enhance it later if need be. Do you agree?

Reply:
Well, I was not talking about a bean class with a producer method but purely a bean class annotated with `@DefaultBean`. From dependency resolution POV there is no difference between a class bean and a producer bean.

Reply:
Yes, that's what I thought you meant as well and TBH I am a little against allowing it (at least for the time being). I think it would create more confusion than it would solve.
Are you OK with us leaving it out for now?